src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -1160,6 +1162,49 @@ static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x,
 
 } // namespace BinaryIntrinsicFunction
 
+
+namespace ObjectType {
+
+     static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x, diag::Diagnostics& diagnostics) {
+        ASRUtils::require_impl(x.n_args == 1,
+            "ASR Verify: type() takes only 1 argument `object`",
+            x.base.base.loc, diagnostics);
+    }
+
+    static ASR::expr_t *eval_ObjectType(Allocator &al, const Location &loc,
+            ASR::ttype_t* t1, Vec<ASR::expr_t*>& /*args*/) {
+        std::string object_type = "<type '" + ASRUtils::type_to_str_python(t1) + "'>";
+        return StringConstant(object_type, character(object_type.length()));
+    }
+
+    static inline ASR::asr_t* create_ObjectType(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        if (args.size() != 1) {
+            err("type() takes exactly 1 argument `object` for now", loc);
+        }
+        ASR::expr_t *m_value = nullptr;
+        Vec<ASR::expr_t *> arg_values;
+       
+        if (ASR::is_a<ASR::IntrinsicScalarFunction_t>(*args[0])) {
+            ASR::IntrinsicScalarFunction_t *object = ASR::down_cast<ASR::IntrinsicScalarFunction_t>(args[0]);
+            if (static_cast<IntrinsicScalarFunctions>(object->m_intrinsic_id) == IntrinsicScalarFunctions::ObjectType) {
+                m_value = StringConstant("<type 'type'>", character(13)); // 13 is the length of the string "<type 'type'>"
```

Comment:
```suggestion
                m_value = StringConstant("<class 'type'>", character(13)); // 13 is the length of the string "<type 'type'>"
```
```py
>>> type(type(1))
<class 'type'>
```

Reply:
I had intentionally not used the word 'class' as we do not currently support it fully. Should we write it here?

Reply:
Nope, we have to make sure both CPython and LPython generates the same output.
So, I suggested the above changes and test this both in CPython and LPython(llvm).

Reply:
I am really sorry for the confusion. The return type of `type()` in CPython is of type - `type`, whereas here we are using a string for it. They would not be similar.

I request guidance on how do we go about this @Thirumalai-Shaktivel.